utils/include/detray/simulation/simulator.hpp

```diff
@@ -98,10 +96,17 @@ struct simulator {
     private:
     config m_cfg;
     std::size_t m_events = 0;
+    std::size_t m_seed = 0;
     std::string m_directory = "";
     std::unique_ptr<detector_t> m_detector;
     track_generator_t m_track_generator;
     smearer_t m_smearer;
+
+    /// Actor states
+    typename parameter_transporter<transform3>::state m_transporter{};
```

Comment:
I guess the simulation is not threadsafe anyway, because of the writer, right? You could also move the ```actor_states``` out of the for loop, to make it clear that the simulator operates every track on the same actor states.

Reply:
It might not be thread safe at all. 
Yeah I just moved it out of the loop